File: DMS_Connector/Business Logic/DataContract/Configuracion/Configuracion Sucursal/Horario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DMS_Connector.Business_Logic.DataContract.Configuracion.Configuracion_Sucursal
{
    /// <summary>
    /// Objeto que representa el horario de una sucursal
    /// </summary>
    public class Horario
    {
        public int MinutosAlmuerzo { get; set; }
        public DateTime HoraInicioAlmuerzo { get; set; }
        public DateTime HoraFinAlmuerzo { get; set; }
        public DateTime HoraApertura { get; set; }
        public DateTime HoraCierre { get; set; }
        public bool HorarioConfigurado { get; set; }

        /// <summary>
        /// Constructor del objeto horario
        /// </summary>
        /// <param name="HoraApertura">Hora de apertura de la sucursal</param>
        /// <param name="HoraCierre">Hora de cierre de la sucursal</param>
        /// <param name="HoraInicioAlmuerzo">Hora de inicio del almuerzo</param>
        /// <param name="HoraFinAlmuerzo">Hora de fin del almuerzo</param>
        public Horario(DateTime HoraApertura, DateTime HoraCierre, DateTime HoraInicioAlmuerzo, DateTime HoraFinAlmuerzo)
        {
            //Se utiliza la fecha mínima, ya que esta es diferente entre los objetos COM y .NET ocasionando
            //que no se puedan verificar de la manera tradicional con DateTime.MinValue
            DateTime FechaMinima;
            TimeSpan EspacioTiempo;
            try
            {
                FechaMinima = new DateTime(1899, 12, 30);
                this.HoraApertura = HoraApertura;
                this.HoraCierre = HoraCierre;
                this.HoraInicioAlmuerzo = HoraInicioAlmuerzo;
                this.HoraFinAlmuerzo = HoraFinAlmuerzo;
                if (!(HoraInicioAlmuerzo == FechaMinima) && !(HoraFinAlmuerzo == FechaMinima))
                {
                    EspacioTiempo = HoraFinAlmuerzo - HoraInicioAlmuerzo;
                    this.MinutosAlmuerzo = Convert.ToInt32(EspacioTiempo.TotalMinutes);
                }

                if (!(HoraApertura == FechaMinima) && !(HoraCierre == FechaMinima))
                {
                    HorarioConfigurado = true;
                }
                else
                {
                    HorarioConfigurado = false;
                }
            }
            catch (Exception ex)
            {
                DMS_Connector.Helpers.ManejoErrores(ex);
            }
        }
    }
}

File: SCG Addon Taller/ServicioAlCliente/Horario.vb
Public Class Horario
    Private Apertura As DateTime
    Private Cierre As DateTime
    Private InicioAlmuerzo As DateTime
    Private FinAlmuerzo As DateTime
    Private DuracionAlmuerzo As Integer = 0
    Private Configurado As Boolean = False

    Public ReadOnly Property MinutosAlmuerzo() As Integer
        Get
            Return DuracionAlmuerzo
        End Get
    End Property

    Public ReadOnly Property HoraInicioAlmuerzo() As DateTime
        Get
            Return InicioAlmuerzo
        End Get
    End Property

    Public ReadOnly Property HoraFinAlmuerzo() As DateTime
        Get
            Return FinAlmuerzo
        End Get
    End Property

    Public ReadOnly Property HoraApertura() As DateTime
        Get
            Return Apertura
        End Get
    End Property

    Public ReadOnly Property HoraCierre() As DateTime
        Get
            Return Cierre
        End Get
    End Property

    Public ReadOnly Property HorarioConfigurado() As Boolean
        Get
            Return Configurado
        End Get
    End Property


    Sub New(ByVal HorarioApertura As DateTime, ByVal HorarioCierre As DateTime, ByVal HorarioInicioAlmuerzo As DateTime, ByVal HorarioFinAlmuerzo As DateTime)
        'Se utiliza la fecha mínima, ya que esta es diferente entre los objetos COM y .NET ocasionando
        'que no se puedan verificar de la manera tradicional con DateTime.MinValue
        Dim FechaMinima As DateTime
        Dim EspacioTiempo As TimeSpan
        Try
            FechaMinima = New DateTime(1899, 12, 30)
            Apertura = HorarioApertura
            Cierre = HorarioCierre
            InicioAlmuerzo = HorarioInicioAlmuerzo
            FinAlmuerzo = HorarioFinAlmuerzo
            If Not InicioAlmuerzo = FechaMinima AndAlso Not FinAlmuerzo = FechaMinima Then
                EspacioTiempo = FinAlmuerzo - InicioAlmuerzo
                DuracionAlmuerzo = EspacioTiempo.TotalMinutes
            End If

            If Not Apertura = FechaMinima AndAlso Not Cierre = FechaMinima Then
                Configurado = True
            End If
        Catch ex As Exception
            DMS_Connector.Helpers.ManejoErrores(ex)
        End Try
    End Sub
End Class
